app/authentication/jti_claim_storage.py

```diff
@@ -0,0 +1,50 @@
+from uuid import UUID
+
+from sqlalchemy.exc import IntegrityError
+from structlog import get_logger
+
+from app.data_model.database import UsedJtiClaim, commit_or_rollback, db_session
+
+logger = get_logger()
+
+
+class JtiTokenUsed(Exception):
+
+    def __init__(self, jti_claim):
+        self.jti_claim = jti_claim
+
+    def __str__(self, *args, **kwargs):
+        return "jti claim '{jti_claim}' has already been used".format(jti_claim=self.jti_claim)
+
+
+def _is_valid(jti_claim):
+    try:
+        UUID(jti_claim, version=4)
+    except ValueError:
+        return False
+    return True
+
+
+def use_jti_claim(jti_claim):
+    """
+    Use a jti claim
+    :param jti_claim: jti claim to mark as used.
+    :raises ValueError: when jti_claim is None.
+    :raises TypeError: when jti_claim is not a valid uuid4.
+    :raises IntegrityError: when jti_claim has already been used.
+    """
+    if jti_claim is None:
+        raise ValueError
+    elif not _is_valid(jti_claim):
```

Comment:
I agree with Chris' first comment on this. I think it should be an `if` rather than an `elif`. As an `elif` when I read it, it makes me glance back to evaluate when it wouldn't get there.